File: Javascript/DataStructures/Stack/myImpl.js
class Stack{

    constructor(){
        this.items = [];
        this.top = 0;
    }

    push(value){
        this.items.push(value);
        this.top++;
    }

    pop(){
        if(this.top != 0){
            return this.items.pop();
            this.top--;
        }else{
            console.log("Stack is Empty !");
        }   
    }

    peek(){
        if (this.top !== 0) {
            return this.items[this.items.length - 1]
        }else{
            console.log("Stack is Empty !");
        }
    }

    getLength(){
        return this.items.length;
        return this.top;
    }

    isEmpty(){
        return this.items.isEmpty();
        return this.top === 0
    }

    static isStack(obj){
        return obj instanceof Stack;
    }
}


let s = new Stack();
s.push(1);
s.push(2);
console.log(s.pop());
console.log(s.peek());

if(s.isEmpty()){
    console.log("Stack is Empty !");
}else{
    console.log("Stack is not Empty");
}

console.log(Stack.isStack(s));

File: Java/DataStructures/Stack/myImpl.java
class Stack {
    static final int MAX = 1000;
    int top;
    int a[] = new int[MAX];

    Stack() {
        top = -1;
    }

    boolean push(int x) {
        if (top >= (MAX - 1)) {
            System.out.println("Index out of bound");
            return false;
        } else {
            a[++top] = x;
            return true;
        }
    }

    int pop() {
        if (top < 0) {
            System.out.println("Underflow");
            return 0;
        } else {
            int x = a[top--];
            return x;
        }
    }

    int peek() {
        if (top < 0) {
            System.out.println("Underflow");
            return 0;
        } else {
            int x = a[top];
            return x;
        }
    }

    boolean isEmpty() {

        if(a.length == 0){
            return true;
        }else{
            return false;
        }
    }
}

class myImpl {
    public static void main() {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.push(30);
        System.out.println(s.pop() + " Popped from stack");
    }
}